pybamm/discretisations/meshes.py

```diff
@@ -104,30 +89,50 @@ def add_ghost_meshes(self):
         ]
         for submeshname, submesh in submeshes:
             edges = submesh.edges
+
             # left ghost cell: two edges, one node, to the left of existing submesh
             lgs_edges = np.array([2 * edges[0] - edges[1], edges[0]])
-            self[submeshname + "_left ghost cell"] = self.submeshclass(lgs_edges)
+            self[submeshname + "_left ghost cell"] = SubMesh(lgs_edges)
+
             # right ghost cell: two edges, one node, to the right of existing submesh
             rgs_edges = np.array([edges[-1], 2 * edges[-1] - edges[-2]])
-            self[submeshname + "_right ghost cell"] = self.submeshclass(rgs_edges)
-
-
-class FiniteVolumeSubmesh:
-    """A submesh for finite volumes.
+            self[submeshname + "_right ghost cell"] = SubMesh(rgs_edges)
 
-    The mesh is defined by its edges; then node positions, diffs and mesh size are
-    calculated from the edge positions.
 
-    Parameters
-    ----------
-    edges : :class:`numpy.array`
-        The position of the edges of the cells
+class SubMesh:
```

Comment:
not sure of the reason behind this class. Looks like this is still assuming a 1D mesh? Should this now go in `Uniform1DSubMesh`? Or should be named `SubMesh1D`?

Reply:
SubMesh1D is probably the most sensible. The combined "negative electrode" "separator" "positive electrode" mesh is an instance of this class but not of Uniform1DSubMesh.